cmd/agent.go

```diff
@@ -175,8 +224,25 @@ func main() {
 			adapter, err = pgprem.CreateDefaultPostgreSQLAdapter()
 		}
 		if err != nil {
-			log.Fatalf("Failed to create adapter: %v", err)
+			fatalWithHelp("Failed to create adapter", err)
 		}
 	}
+
+	// Sanity check various functions of the adapter
+	log.Println("Checking adapter can get system information...")
+	_, err = adapter.GetSystemInfo()
+	if err != nil {
+		fatalWithHelp("Failed to get database system information", err)
+	}
+	log.Println("Successfully retrieved system information")
+
+	log.Println("Checking adapter can fetch metrics...")
+	_, errors := adapter.GetMetrics()
+	if errors != nil {
+		fatalWithHelpMultipleErrors("Failed to get metrics from database", errors)
+	}
+	log.Println("Successfully fetched metrics")
+
```

Comment:
Is it worth checking that we can get the config as well? Depending on the adapter this will test a different thing than the metrics or system info fetch I think

More generally, instead of doing these here, should the be incorporated into the runner? So every runner loop that you set up _has_ a start up check - Just sort of imagining if we have autovacuum loops, different frequency collections etc how do we scale this out?